Fedora build problems

On Fedora 28 : 

CC=/usr/bin/clang CXX=/usr/bin/clang++ ./build.sh debug

... works (though bluegl doesn't work on 32 bit systems).

If I do : 

./build.sh debug

... I get "Could NOT find Threads (missing: Threads_FOUND)" .  Dunno why.

Also, if I do

./build.sh -m debug

... I get : c++: error: unrecognized command line option ‘-stdlib=libc++’

This is GNU 8.1.1

I don't see pthread-dev . I see /usr/include/pthread.h , which is part of glibc-headers

clang is 6.0.1

Here's the output of ./build.sh debug : 

Warning: JAVA_HOME is not set, skipping Java projects
Building debug in out/cmake-debug...
-- The C compiler identification is GNU 8.1.1
-- The CXX compiler identification is GNU 8.1.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - not found
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find Threads (missing: Threads_FOUND)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake/Modules/FindThreads.cmake:205 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  third_party/libgtest/tnt/CMakeLists.txt:26 (find_package)


-- Configuring incomplete, errors occurred!
See also "~/filament/out/cmake-debug/CMakeFiles/CMakeOutput.log".
See also "~/filament/out/cmake-debug/CMakeFiles/CMakeError.log".

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,13 +3,6 @@
 # ==================================================================================================
 cmake_minimum_required(VERSION 3.1)
 
-if (WIN32)
-    # On Windows we need to instruct cmake to generate the .def in order to get the .lib required
-    # when linking against dlls. CL.EXE will not generate .lib without .def file (or without pragma
-    # __declspec(dllexport) in front of each functions).
-    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
-endif()
-
 # ==================================================================================================
 # Project declaration
 # ==================================================================================================
@@ -29,6 +22,11 @@ if (UNIX AND NOT APPLE AND NOT ANDROID)
 endif()
 
 if (WIN32)
+    # On Windows we need to instruct cmake to generate the .def in order to get the .lib required
+    # when linking against dlls. CL.EXE will not generate .lib without .def file (or without pragma
+    # __declspec(dllexport) in front of each functions).
+    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
     # TODO: Figure out why pdb generation messes with incremental compilaton.
     # IN RELEASE_WITH_DEBUG_INFO, generate debug info in .obj, no in pdb.
     set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} /Z7")
@@ -98,6 +96,16 @@ endif()
 if (UNIX AND CMAKE_GENERATOR STREQUAL "Ninja")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fcolor-diagnostics")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fcolor-diagnostics")
+endif()
+
+# ==================================================================================================
+# Compiler check
+# ==================================================================================================
+if ("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    message(FATAL_ERROR
+            "The GCC compiler is not supported, please use clang for both C and C++. "
+            "Please refer to README.md for more information.")
+    return()
 endif()
 
 # ==================================================================================================

